internal/dao/sqlx/source_dao.go

```diff
@@ -0,0 +1,124 @@
+	getSourceById    = `SELECT * FROM sources WHERE id = $1;`
+	listSources      = `SELECT * FROM sources ORDER BY id LIMIT $1 OFFSET $2`
+	deleteSourceById = `DELETE FROM sources WHERE id = $1 RETURNING source_id`
+	getSourceByOrgId = `SELECT * FROM sources WHERE org_id = $1 ORDER BY id LIMIT 1;`
+)
+
+type sourceDaoSqlx struct {
+	name       string
+	getById    *sqlx.Stmt
+	list       *sqlx.Stmt
+	create     *sqlx.Stmt
+	getByOrgId *sqlx.Stmt
+	deleteById *sqlx.Stmt
+}
+
+func getSourceDao(ctx context.Context) (dao.SourceDao, error) {
+	var err error
+	daoImpl := sourceDaoSqlx{}
+	daoImpl.name = "source"
+
+	daoImpl.getById, err = db.DB.PreparexContext(ctx, getSourceById)
+	if err != nil {
+		return nil, NewPrepareStatementError(ctx, &daoImpl, getSourceById, err)
+	}
+	daoImpl.list, err = db.DB.PreparexContext(ctx, listSources)
+	if err != nil {
+		return nil, NewPrepareStatementError(ctx, &daoImpl, listSources, err)
+	}
+	daoImpl.create, err = db.DB.PreparexContext(ctx, createSource)
+	if err != nil {
+		return nil, NewPrepareStatementError(ctx, &daoImpl, createSource, err)
+	}
+	daoImpl.getByOrgId, err = db.DB.PreparexContext(ctx, getSourceByOrgId)
+	if err != nil {
+		return nil, NewPrepareStatementError(ctx, &daoImpl, getSourceByOrgId, err)
+	}
+	return &daoImpl, nil
+}
+
+func (dao *sourceDaoSqlx) NameForError() string {
+	return dao.name
+}
+
+func init() {
+	dao.GetSourceDao = getSourceDao
+}
+
+func (dao *sourceDaoSqlx) Create(ctx context.Context, source *models.Source) (*models.Source, error) {
+	query := createSource
+	stmt := dao.create
+	err := stmt.GetContext(ctx, source, source.SourceID, source.Name, source.OrgID, source.AuthID)
+	if err != nil {
+		return nil, NewGetError(ctx, dao, query, err)
+	}
+	return source, nil
+}
+
+func (dao *sourceDaoSqlx) GetById(ctx context.Context, id uint64) (*models.Source, error) {
+	query := getSourceById
+	stmt := dao.getById
+	result := &models.Source{}
+
+	err := stmt.GetContext(ctx, result, id)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, NewNoRowsError(ctx, dao, query)
+		} else {
+			return nil, NewGetError(ctx, dao, query, err)
+		}
+	}
+	return result, nil
+}
+
+func (dao *sourceDaoSqlx) List(ctx context.Context, limit uint64, offset uint64) ([]*models.Source, error) {
+	query := listSources
+	stmt := dao.list
+	var result []*models.Source
+
+	err := stmt.SelectContext(ctx, &result, limit, offset)
+	if err != nil {
+		return nil, NewSelectError(ctx, dao, query, err)
+	}
+	return result, nil
+}
+
+func (dao *sourceDaoSqlx) GetByOrgId(ctx context.Context, orgId string) (*models.Source, error) {
+	query := getSourceByOrgId
+	stmt := dao.getByOrgId
+	result := &models.Source{}
+
+	err := stmt.GetContext(ctx, result, orgId)
+	if err != nil {
+		return nil, NewGetError(ctx, dao, query, err)
+	}
+	return result, nil
+}
+
+func (dao *sourceDaoSqlx) Delete(ctx context.Context, id uint64) (*models.Source, error) {
+	query := deleteSourceById
+	stmt := dao.deleteById
```

Comment:
This will fail.

Reply:
It does fail, not sure why. can you explain? how is it different from the other function?